src/core/config.c

```diff
@@ -0,0 +1,133 @@
+#include "config.h"
+#include "xdpw.h"
+#include "logger.h"
+
+#include <dictionary.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <wordexp.h>
+#include "iniparser.h"
+
+void print_config(enum LOGLEVEL loglevel, struct xdpw_config *config) {
+	logprint(loglevel, "config: Configfile %s",config->conf.configfile);
+	logprint(loglevel, "config: Loglevel %d",config->logger_conf.loglevel);
+	logprint(loglevel, "config: Outputname  %s",config->screencast_conf.output_name);
+}
+
+// NOTE: calling destroy_config won't prepare the config to be read again from config file
+// with init_config since to pointers and other values won't be reset to NULL, or 0
+void destroy_config(struct xdpw_config *config) {
+	free(&config->conf.configfile);
+
+	// screencast
+	free(&config->screencast_conf.output_name);
+}
+
+void getstring_from_conffile(dictionary *d, char *key, char **dest, char *def) {
+	if (*dest != NULL) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, def);
+	if (c != NULL) {
+		// Allow keys without value as default
+		if (strcmp(c, "") != 0) {
+			*dest = strdup(c);
+		} else {
+			*dest = strdup(def);
+		}
+	}
+}
+
+void getenum_from_conffile(dictionary *d, char *key, int *dest, const char *array[] , int num_array, int def) {
+	if (*dest != 0) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, NULL);
+	if (c != NULL) {
+		for (int i = 0; i < num_array; i++) {
+			if (strcmp(c, array[i]) == 0) {
+				*dest = i;
+				return;
+			}
+		}
+	}
+	*dest = def;
+}
+
+static bool file_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+static char *expand_path(char *path, bool realloc) {
+	if (path != NULL) {
+		char *expand = (char*)calloc(strlen(path)+3, sizeof(char));
+		// Quote string to preserve whitespaces
```

Comment:
This breaks paths which contain quotes.

I really don't think we should be using `wordexp` here.

Reply:
can't we just use the original version, since we only apply this function to hardcoded strings and the only point of error afaics is if someone has a space or another problematic special character as SYSCONFDIR

Reply:
The original version breaks when any expanded variable contains a space. I don't think that's reasonable.

Reply:
Yeah makes sense, sry I thought wordexp would only seperate strings on spaces in the to expand string and not in the expanded string and didn't tested it. 

I changed it to assemble the path directly from environment variables.

Reply:
Can I resolv this with the new implementation via getenv?

Reply:
Yeah sure, the new implementation is completely different.